src/wallet/wallet2.cpp

```diff
@@ -692,10 +692,11 @@ void wallet2::expand_subaddresses(const cryptonote::subaddress_index& index)
 //----------------------------------------------------------------------------------------------------
 std::string wallet2::get_subaddress_label(const cryptonote::subaddress_index& index) const
 {
-  if (index.major >= m_subaddress_labels.size())
-    throw std::runtime_error("index.major is out of bound");
-  if (index.minor >= m_subaddress_labels[index.major].size())
-    throw std::runtime_error("index.minor is out of bound");
+  if (index.major >= m_subaddress_labels.size() || index.minor >= m_subaddress_labels[index.major].size())
+  {
+    MERROR("Subaddress label doesn't exist");
+    return "";
+  }
```

Comment:
Please also update the `set_subaddress_label` in the same way. Also the comments in wallet2.h should be removed for consistency:

https://github.com/monero-project/monero/blob/8d511f3c2486d22400cdd69c8d3640ee2d20aeff/src/wallet/wallet2.h#L471

Also, there are a few calls to `createTransaction` in tests/libwallet_api_tests/main.cpp which should be adapted too.

Reply:
@kenshi84 done